src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs

```diff
@@ -35,8 +46,19 @@ protected internal class State
 
             public bool IsConstructorInitializerGeneration { get; private set; }
 
-            private State()
+            public IMethodSymbol DelegatedConstructor { get; private set; }
+
+            public ImmutableDictionary<string, ISymbol> ParameterToExistingMemberMap { get; private set; }
+            public ImmutableDictionary<string, string> ParameterToNewFieldMap { get; private set; }
+            public ImmutableDictionary<string, string> ParameterToNewPropertyMap { get; private set; }
+            public ImmutableArray<IParameterSymbol> RemainingParameters { get; private set; }
```

Comment:
these values moved in here from the editor.  instead of waiting till the applicaitno to compute them, we compute up front.  That way we know what options to offer the user (i.e. if we can offer to generate fields/properties).